File: wraps/python/src/indicators/common/Quote.py
from Skender.Stock.Indicators import Quote as CsQuote
from cstypes import DateTime, Decimal

class Quote(CsQuote):
    def __init__(self, date, open = None, high = None, low = None, close = None, volume = None):
        self.Date = DateTime(date)
        self.Open = Decimal(open)
        self.High = Decimal(high)
        self.Low = Decimal(low)
        self.Close = Decimal(close)
        self.Volume = Decimal(volume)

File: src/_common/Quotes/Quote.cs
namespace Skender.Stock.Indicators;

/// <summary>
/// Built-in Quote type, representing an OHLCV aggregate price period.
/// </summary>
/// <param name="Timestamp">
/// Close date/time of the aggregate
/// </param>
/// <param name="Open">
/// Aggregate bar's first tick price
/// </param>
/// <param name="High">
/// Aggregate bar's highest tick price
/// </param>
/// <param name="Low">
/// Aggregate bar's lowest tick price
/// </param>
/// <param name="Close">
/// Aggregate bar's last tick price
/// </param>
/// <param name="Volume">
/// Aggregate bar's tick volume
/// </param>
/// <inheritdoc cref="IQuote"/>
[Serializable]
public record Quote
(
    DateTime Timestamp,
    decimal Open,
    decimal High,
    decimal Low,
    decimal Close,
    decimal Volume
) : IQuote
{
    /// <inheritdoc/>
    [JsonIgnore]
    public double Value => (double)Close;

    /// <inheritdoc/>
    [Obsolete("Use 'Timestamp' property instead.")]
    public DateTime Date
    {
        get => Timestamp;
        init => Timestamp = value;
    }

    // TODO: this can be removed when Date is removed
    // It allows old `new Quote { Date: ... }` initialization.
    /// <inheritdoc/>
    public Quote()
        : this(default, default, default, default, default, default) { }
}
